src/std/rpc/ws_client/mod.rs

```diff
@@ -15,35 +14,37 @@ use ac_node_api::Events;
    limitations under the License.
 
 */
+
 pub use ac_node_api::{events::EventDetails, StaticEvent};
 
-use crate::{
-	std::{
-		error::Error, json_req, rpc::RpcClientError, Api, ApiResult, FromHexString,
-		RpcClient as RpcClientTrait, XtStatus,
-	},
-	utils, Hash, Index,
-};
-use ac_node_api::DispatchError;
-use ac_primitives::ExtrinsicParams;
+use crate::std::{rpc::RpcClientError, XtStatus};
 use log::*;
 use serde_json::Value;
-use sp_core::Pair;
-use sp_runtime::MultiSignature;
 use std::{
 	fmt::Debug,
-	sync::mpsc::{Receiver, SendError, Sender as ThreadOut},
+	sync::mpsc::{SendError, Sender as ThreadOut},
 };
 use ws::{CloseCode, Error as WsError, Handler, Handshake, Message, Result as WsResult, Sender};
 
 pub use client::WsRpcClient;
+pub use subscription::*;
 
 pub mod client;
+pub mod subscription;
 
 type RpcResult<T> = Result<T, RpcClientError>;
 
 pub type RpcMessage = RpcResult<Option<String>>;
 
+#[allow(clippy::result_large_err)]
+pub trait Subscriber {
+	fn start_subscriber(
+		&self,
+		json_req: String,
+		result_in: ThreadOut<String>,
+	) -> Result<(), WsError>;
+}
+
```

Comment:
Why didn't you move this one to the subscription module?

Reply:
Good question. Original thought: Because the api does not implement the subscription trait. But that means, the api is still in the wrong place.. Should not be in the ws_client folder. Let me think on that.

Reply:
Better now?

Added a "goal" to the PR description, which hopefully makes it clear what I want out of this PR.